File: backend/src/main/java/com/todo/backend/service/TaskService.java
package com.todo.backend.service;

import com.todo.backend.enums.ResponseCode;
import com.todo.backend.model.request.TaskRequest;
import com.todo.backend.model.response.TaskResponse;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface TaskService {

    ResponseCode addTask(TaskRequest taskRequest);

    ResponseCode updateTask(TaskRequest taskRequest, Integer taskId);

    List<TaskResponse> getTaskList(Integer taskMapId, String sortBy, String sortDir, String searchBy, String searchParameter);

}

File: frontend/src/services/taskService.ts
import { axiosClient } from "../config/axios";
import { Filter } from "../interface/filterInterface";
import { Task } from "../interface/taskInterface";
import { errorHandler } from "./errorHandler";

class TaskService {
  async addTask(payload: Task) {
    const response = await axiosClient.post("/addTask", {
      taskName: payload.name,
      taskDescription: payload.description,
      taskStatus: payload.status,
      taskMapId: payload.taskMapId,
    });
    return errorHandler(response);
  }
  async updateTask(payload: Task) {
    const response = await axiosClient.post(`/updateTask/${payload.taskId}`, {
      taskName: payload.name,
      taskDescription: payload.description,
      taskStatus: payload.status,
    });
    return errorHandler(response);
  }
  async getTaskList(payload: Filter) {
    const response = await axiosClient.get(
      `/getTaskList/${payload.taskMapId}?sortBy=${payload.sortBy}&sortDir=${payload.sortDir}&searchBy=${payload.searchBy}&searchParameter=${payload.searchParamter}`
    );
    return errorHandler(response);
  }
}
export const taskService = new TaskService();
